### PR TITLE
Restore backup configuration to Cloud Shell

### PR DESCRIPTION
**Pull request by louisjimenez**

### CODE REVIEW

File: scripts/manage/landing_page_base.md [hunk 1]
```diff
@@ -179,6 +179,32 @@ Update the commands and documentation in your management environment to the late
 ~/spinnaker-for-gcp/scripts/manage/connect_to_redis.sh
 ```
 
+### Restore a backup to Cloudshell
```
Comment (duftler) [inline scripts/manage/landing_page_base.md]: s/Cloudshell/Cloud Shell

here and in various other spots

File: scripts/manage/landing_page_base.md [hunk 2]
```diff
@@ -179,6 +179,32 @@ Update the commands and documentation in your management environment to the late
...
+
+Restore a backup of the halyard configuration and deployment configuration from Cloud Source Repositories to your Cloudshell. 
+
+```bash
+~/spinnaker-for-gcp/scripts/manage/restore_backup_to_cloudshell.sh -p $PROJECT_ID -r $CONFIG_CSR_REPO -h GIT_HASH
+```
+
+To apply the halyard config changes to the cluster:
```
Comment (duftler) [inline scripts/manage/landing_page_base.md]: Looks like `push_and_apply` and `setup` are both referenced when the `restore_backup_to_cloudshell` scripts completes. Do we need the instructions in both locations?

I'd lean towards dropping them from the markdown unless you feel strongly about having them in both locations.

Comment (louisjimenez) [reply scripts/manage/landing_page_base.md]: Agreed, I will remove it from the markdown.

File: scripts/manage/restore_backup_to_cloud_shell.sh [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+EXAMPLE_COMMAND="'restore_backup_to_cloud_shell.sh -p PROJECT -r REPOSITORY_NAME -h GIT_HASH'"
+
+if [ -z "$PROJECT_ID" ]; then
+  bold "Project id is required. $EXAMPLE_COMMAND"
+  exit 1
+fi
+
+if [ -z "$CONFIG_CSR_REPO" ]; then
+  bold "Cloud Source repository name is required. $EXAMPLE_COMMAND"
```
Comment (duftler) [inline scripts/manage/restore_backup_to_cloud_shell.sh]: We have `Cloud Source repository` here, but `Cloud source repository` down below. I'd go with `Cloud Source Repository` for both.

File: scripts/manage/landing_page_base.md [hunk 4]
```diff
@@ -179,6 +179,20 @@ Update the commands and documentation in your management environment to the late
...
 
+### Restore a backup to Cloud Shell
+
+Restore a backup of the halyard configuration and deployment configuration from Cloud Source Repositories to your Cloud Shell. 
+
+```bash
+~/spinnaker-for-gcp/scripts/manage/restore_backup_to_cloud_shell.sh -p $PROJECT_ID -r $CONFIG_CSR_REPO -h GIT_HASH
+```
+
+All backups can be viewed by running:
```
Comment (duftler) [inline scripts/manage/landing_page_base.md]: Can we just include the CSR link directly in the markdown? The landing_page_expanded.md file is generated anyway and you're using the template variables just above already. Seems a shorter path than running the script and then clicking on the link.

File: scripts/manage/restore_backup_to_cloud_shell.sh [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+while getopts ":p:r:h:" options; do
+  case $options in
+    p ) PROJECT_ID=$OPTARG ;;
+    r ) CONFIG_CSR_REPO=$OPTARG ;;
+    h ) GIT_HASH=$OPTARG ;;
+    \? ) bold "invalid option supplied: -$OPTARG"
+  esac
+done
+
+EXAMPLE_COMMAND="'restore_backup_to_cloud_shell.sh -p PROJECT -r REPOSITORY_NAME -h GIT_HASH'"
```
Comment (duftler) [inline scripts/manage/restore_backup_to_cloud_shell.sh]: If I just copy/paste and run this command, there is no indication that `GIT_HASH` wasn't recognized as an actual commit hash. Same if I enter any other random string in place of a legitimate hash. We should confirm that the hash refers to an actual commit and print an error message if it does not.

File: scripts/manage/restore_backup_to_cloud_shell.sh [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+mkdir -p ~/.spin
+remove_and_copy config ~/.spin/config
+remove_and_copy key.json ~/.spin/key.json
+
+popd
+rm -rf $TEMP_DIR
+
+bold "Configuration applied. To diff this config with what was last deployed, go to:"
+bold "https://source.cloud.google.com/$PROJECT_ID/$EXISTING_CSR_REPO/+/$GIT_HASH...master"
+bold "To apply the halyard config changes to the cluster, run ~/spinnaker-for-gcp/scripts/manage/push_and_apply.sh. To apply changes in the properties file to your deployment, run ~/spinnaker-for-gcp/scripts/install/setup.sh"
```
Comment (duftler) [inline scripts/manage/restore_backup_to_cloud_shell.sh]: Can we break this up into multiple lines? Would like to be able to clearly see and capture the `~/spinnaker-for-gcp/scripts/manage/push_and_apply.sh` command.

File: scripts/manage/landing_page_base.md (hunk 4 above)
Comment (louisjimenez) [reply scripts/manage/landing_page_base.md]: Good idea. This script served more of a purpose when we were retrieving and displaying a listing from storage buckets but I agree, just putting it in the landing page doc is a better UX.

File: scripts/manage/landing_page_base.md [hunk 7]
```diff
@@ -179,6 +179,16 @@ Update the commands and documentation in your management environment to the late
...
 
+### Restore a backup to Cloud Shell
+
+Restore a backup of the halyard configuration and deployment configuration from Cloud Source Repositories to your Cloud Shell. 
+
+```bash
+~/spinnaker-for-gcp/scripts/manage/restore_backup_to_cloud_shell.sh -p $PROJECT_ID -r $CONFIG_CSR_REPO -h GIT_HASH
+```
+
+All backups can be viewed in [Cloud Source Repository](https://source.cloud.google.com/$PROJECT_ID/$CONFIG_CSR_REPO).
```
Comment (duftler) [inline scripts/manage/landing_page_base.md]: s/in/in this

**Pull request merged**